### PR TITLE
feat: initial setup for sybase

### PR DESCRIPTION
**Pull request by fangxuyi**
### SUMMARY

add connectors for sybase

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -157,6 +157,7 @@ def get_git_sha() -> str:
         "snowflake": [
             "snowflake-sqlalchemy==1.2.4"
         ],  # PINNED! 1.2.5 introduced breaking changes requiring sqlalchemy>=1.4.0
+        "sybase": ["sqlalchemy-sybase>=1.0.6"],
```
Comment (villebro) [inline setup.py:160]: nit: Let's restrict this to the current major version:
```suggestion
        "sybase": ["sqlalchemy-sybase>=1.0.6, <2"],
```

File: superset/db_engine_specs/sybase.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            return f"""DATETIME '{dttm_formatted}'"""
+        if tt == utils.TemporalType.TIMESTAMP:
+            dttm_formatted = dttm.isoformat(timespec="microseconds")
+            return f"""TIMESTAMP '{dttm_formatted}'"""
+        return None
+
+    @classmethod
+    def alter_new_orm_column(cls, orm_col: "TableColumn") -> None:
+        if orm_col.type == "TIMESTAMP":
+            orm_col.python_date_format = "epoch_ms"
```
Comment (villebro) [inline superset/db_engine_specs/sybase.py:64]: Is this really required? This is only needed if the database returns timestamps as numeric values, in this case millisecond epochs. So far I've only seen this being needed in the CrateDB spec.

**Review by villebro — COMMENTED** [review]
Thanks so much for this contribution! The changes look great, but I left a few small comments. Also, it would be great if you could add the following to this PR:
1. Add a few lines to the documentation on how to connect to a Sybase instance. The main page with a table of the supported databases lies under `docs/src/pages/Connecting to Databases/index.mdx`. To create a new entry for Sybase, just add a line to the table, and then create a new page `sybase.mdx` by copying one of the existing pages. For instance, the documentation for Drill (`drill.mdx`) is quite comprehensive and serves as a good template.
2. Add a few unit tests in a new file `test_sybase.py` under `tests/unit_tests/db_engine_specs`. Again, you can just copy one of the existing files and modify it accordingly.

If you need help with any of the steps above let us know, as we're happy to help out!

File: superset/db_engine_specs/sybase.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        "P1Y": "DATEPART(YEAR, {col})",
+    }
+
+    @classmethod
+    def epoch_ms_to_dttm(cls) -> str:
+        return "{col}"
+
+    @classmethod
+    def convert_dttm(cls, target_type: str, dttm: datetime) -> Optional[str]:
+        print(cls)
```
Comment (geido) [inline superset/db_engine_specs/sybase.py:49]:
```suggestion
```

**Review by geido — COMMENTED** [review]
Awesome first contribution! Also happy to help with moving this PR forward with tests and documentation

**Pull request closed without merging**